### ISSUE TITLE
Possible Missed Meal Alert Triggering at invalid time

### ISSUE THREAD

**Issue by ELUTE** [LoopKit/Loop#1972]
**Describe the bug**
Missed Meal Alert is triggering when user has a lot of IOB and has bolused for a meal but is dropping significantly. I would expect it not to trigger at this time but rather when bg is rising without a bolus. 

**Attach an Issue Report**
Attached

**To Reproduce**
Turn on Missed Meal Alert

**Expected behavior**
Expect the alert to trigger when BG is rising and no meal bolus is present. 

**Screenshots**
Added

**Phone**
 - Hardware: iPhone 11
 - OS Version: 16.3.1

**Loop Version**
 - Version Number: 3.2.1
 - Repo: LoopKit/Loop

**CGM**
 - Device: Dexcom G6 
 - Manager app: Dexcom App (mmol)

**Pump**
 - Manufacturer: Medtronic
 - Model: 754
 - Firmware version: 2.7

**Additional context**
[Loop Report 2023-03-30 15:24:13-04:00.md](https://github.com/LoopKit/Loop/files/11115391/Loop.Report.2023-03-30.15.24.13-04.00.md)
![IMG_1037](https://user-images.githubusercontent.com/7684327/228944299-27a10c07-6145-4373-b852-e21ea1745052.jpeg)
![IMG_1038](https://user-images.githubusercontent.com/7684327/228944335-1d00885a-425a-4009-ad63-40461437934e.jpeg)
![IMG_1039](https://user-images.githubusercontent.com/7684327/228944354-d4c5ae08-1682-4900-b509-f1bdd4636771.jpeg)

**Comment by ELUTE** [comment]
Additional Incident
[Loop Report 2023-03-28 13:58:52
![IMG_1020](https://user-images.githubusercontent.com/7684327/228946442-f3e96fdf-8aea-45b9-bb6a-2d20f8e1c109.jpeg)
-04:00.md](https://github.com/LoopKit/Loop/files/11115459/Loop.Report.2023-03-28.13.58.52-04.00.md)

![IMG_1021](https://user-images.githubusercontent.com/7684327/228946475-799cf56b-2344-415a-87b9-0af2d30a6697.jpeg)

**Comment by elnjensen** [comment]
Hi Kate - 

I think I see what could be going on here.  In both cases, you show meals with a substantial amount of carbs that are entered with a 30-minute absorption time, but which weren't absorbed that quickly.  I believe that [Loop uses a minimum absorption rate that will only keep carbs active for up to 1.5x the entered absorption time](https://loopkit.github.io/loopdocs/operation/algorithm/prediction/#minimum-carbohydrate-absorption-rate).  So if all of the entered carbs don't show up within 45 minutes, then Loop removes them from being active carbs, and assumes that either they were never eaten, or that they were miscounted.  

Meanwhile, the insulin is still active.  And the carbs are (presumably) really still there, just absorbing more slowly.  So Loop sees that (a) there is active IOB, (b) there are no active carbs to speak of, and (c) that BG isn't dropping much.  That translates into assuming (correctly, as it turns out) that there are carbs being absorbed that aren't included in the COB value.  That's what triggers the "missed meal" indicator - more COB than Loop can account for.  

I would guess that if you increased the absorption time for those meals, you wouldn't get that notification.

**Comment by novalegra** [comment]
I just took a look at the report and can confirm that the short-duration entries aren't being observed in the carb effects (due to the short duration time & them expiring), so when Loop attempts to look at the unexpected deviation it thinks there was a positive difference that it didn't expect. The meal detector is using Loop's normal carb effects, so really fixing the root of this issue (too-short carb duration entered) would probably require hard-coding something in the meal detector related to not sending notifications within X timeframe of entering carbs, which I'm concerned wouldn't correctly handle other cases.

Have you noticed the issue continuing to happen when longer carb durations are entered? I noticed her ICE-based threshold is on the lower side, so I'm also wondering if she might be more susceptible to getting false-positives because of CGM calibrations or noise
